server/src/main/java/com/epam/aidial/core/server/service/ResourceOperationService.java

```
@@ -76,10 +85,47 @@ public void moveResource(ResourceDescriptor source, ResourceDescriptor destinati
             }
         }
 
-        if (destination.getType() == ResourceTypes.APPLICATION) {
+        if (destination.getType() == APPLICATION) {
             applicationService.deleteApplication(source, EtagHeader.ANY);
         } else {
             resourceService.deleteResource(source, EtagHeader.ANY);
         }
     }
+
+    public boolean deleteResource(ResourceDescriptor resource, EtagHeader etag, boolean cleanupLinks) {
+        verifyResourceToDelete(resource);
+        MutableObject<Boolean> deleted = new MutableObject<>();
+        if (cleanupLinks) {
+            String bucketName = resource.getBucketName();
+            String bucketLocation = resource.getBucketLocation();
+            // links to dependent resources should be removed under user's bucket lock
+            lockService.underBucketLock(bucketLocation, () -> {
+                invitationService.cleanUpResourceLink(bucketName, bucketLocation, resource);
+                shareService.revokeSharedResource(bucketName, bucketLocation, resource);
+                deleted.setValue(deleteResourceInternally(resource, etag));
+                return null;
+            });
+        } else {
+            deleted.setValue(deleteResourceInternally(resource, etag));
+        }
+        return deleted.getValue();
+    }
+
+    private static void verifyResourceToDelete(ResourceDescriptor resource) {
+        ResourceType type = resource.getType();
+        if (!(APPLICATION == type || FILE == type
+                || CONVERSATION == type || type == PROMPT)) {
+            throw new IllegalArgumentException("Unsupported resource type to delete: " + type.name());
+        }
+    }
+
+    private boolean deleteResourceInternally(ResourceDescriptor resource, EtagHeader etag) {
+        if (resource.getType() == APPLICATION) {
+            applicationService.deleteApplication(resource, etag);
```

Should you catch ResourceNotFound and return false in this case?

---

I would make a consistent behavior either we throw exception or return false on resource not found.

it turns out resource service returns false but application service throws an exception for the same case

---

fixed